test/parser/test_abstract_parser.py

```diff
@@ -0,0 +1,38 @@
+#
+# Shoulder
+# Copyright (C) 2018 Assured Information Security, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import unittest
+
+from shoulder.parser.abstract_parser import AbstractParser
+
+class TestAbstractParser(unittest.TestCase):
+
+    def test_parser_properties(self):
+        a = AbstractParser
+        a.__abstractmethods__ = frozenset()
```

Comment:
eeeeew

This feels a bit of an...intrusive test, and I don't really see the point. Maybe we should just test that instantiating it raises `TypeError`?

Reply:
To go even further than your suggestion: I will start adding abstract classes to the exclusions list in the unit test coverage. You're right, this is horrifying, and I don't want to write this test over and over for all abstract base classes.